src/FixedRateIrm.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {IIrm} from "../lib/morpho-blue/src/interfaces/IIrm.sol";
+import {IFixedRateIrm} from "./interfaces/IFixedRateIrm.sol";
+
+import {MarketParamsLib} from "../lib/morpho-blue/src/libraries/MarketParamsLib.sol";
+import {Id, MarketParams, Market} from "../lib/morpho-blue/src/interfaces/IMorpho.sol";
+
+/* ERRORS */
+
+/// @dev Thrown when the rate is not already set for this market.
+string constant RATE_NOT_SET = "rate not set";
+/// @dev Thrown when the rate is already set for this market.
+string constant RATE_SET = "rate set";
+/// @dev Thrown when trying to set the rate at zero.
+string constant RATE_ZERO = "rate zero";
+
+/// @title FixedRateIrm
+/// @author Morpho Labs
+/// @custom:contact security@morpho.org
+contract FixedRateIrm is IFixedRateIrm {
+    using MarketParamsLib for MarketParams;
+
+    /* EVENTS */
+
+    /// @notice Emitted when a borrow rate is set.
+    event SetBorrowRate(Id indexed id, uint256 newBorrowRate);
+
+    /* STORAGE */
+
+    /// @notice Borrow rates.
+    mapping(Id => uint256) public borrowRateStored;
+
+    /* SETTER */
+
+    /// @notice Sets the borrow rate for a market.
+    /// @dev A rate can be set by anybody, but only once.
+    /// @dev `borrowRate` reverts on rate not set, so the rate needs to be set before the market creation.
+    /// @dev The creator of a market with this IRM would typically batch setting of the rate with the market creation.
```

Comment:
```suggestion
    /// @dev Batch the setting of the rate with the market creation to ensure that the market is using the intended rate.
```
This explains better why it is advised to batch: if there can be some delay between those 2 actions, then it can be that the first one was not included onchain and that the market uses an incorrect rate

Reply:
actually now the rate must be set at market creation so it's not longer really necessary to batch with the market creation, but more with the oracle creation actually

Reply:
How is the rate set at market creation ?

Reply:
> How is the rate set at market creation ?

I guess using a multicall

Reply:
But this is not enforced right ?

Reply:
No, per the PR's description: it is the market creator's responsibility

Reply:
> actually now the rate must be set at market creation so it's not longer really necessary to batch with the market creation, but more with the oracle creation actually

It's not necessary, but it ensures that your market is using the intended rate. Otherwise the rate could have been set by someone else, your transaction setting the rate would fail but not the transaction creating the market. This is why it's good to batch those (to benefit from the eventual revert of the rate setting, if it was already set)

Reply:
should we put this ?
```suggestion
    /// @dev The creator of a market with this IRM would typically batch setting of the rate with the oracle deployment.
```

Reply:
Oracle deployment is not always necessary to create a market (and so to set its rate), notably when creating a market that differs from another just by its LLTV. Besides I think the most important point is not that an oracle has been deployed for nothing (you could just redeploy it), but that a market has been created with an incorrect rate without noticing it (you don't control the liquidity that is deposited there at that point)

Reply:
prop
```suggestion
```

Reply:
Why removing it?

Reply:
It's outdated. As market creation now revert on not yet set rate. So you don't have to batch the set of the rate and the market creation.